File: todo_work/models/todo.py
__all__ = ["Todo"]


from django.db import models
from users.models import CustomUser
from todo_work.models import Project


class Todo(models.Model):
    project = models.ForeignKey(Project, on_delete=models.CASCADE)
    text = models.TextField()
    created = models.DateTimeField(auto_now_add=True, editable=False)
    updated = models.DateTimeField(auto_now=True, editable=False)
    user = models.ForeignKey(CustomUser, models.PROTECT)
    is_active = models.BooleanField(default=True)

File: frontend/src/components/Todo.js
import React from 'react'


const TodoItem = ({todo}) => {
    return (
        <tr>
            <td>
                {todo.project}
            </td>
            <td>
                {todo.text}
            </td>
            <td>
                {/* {todo.user.username} */}
                {todo.user}
            </td>
            <td>
                {todo.created}
            </td>
            <td>
                {todo.updated}
            </td>
            <td>
                {todo.is_active}
            </td>
        </tr>
    )
}

const TodoList = ({todos}) => {
    return (
        <div>
            <h2>Todo</h2>
            <table>
                <th>
                    Project
                </th>
                <th>
                    Text
                </th>
                <th>
                    User
                </th>
                <th>
                    Created
                </th>
                <th>
                    Updated
                </th>
                <th>
                    Is_active
                </th>
                {todos.map((todo) => <TodoItem todo={todo} />)}
            </table>
        </div>
    )
}
export default TodoList